lightning/src/ln/channel.rs

```diff
@@ -3537,112 +3643,36 @@ impl<Signer: Sign> Channel<Signer> {
+			let mut funding_tx_confirmations = height as i64 - self.funding_tx_confirmation_height as i64 + 1;
+			if self.funding_tx_confirmation_height == 0 {
+				// Note that check_get_funding_locked may reset funding_tx_confirmation_height to
+				// zero if it has been reorged out, however in either case, our state flags
+				// indicate we've already sent a funding_locked
+				funding_tx_confirmations = 0;
 			}
-		}
 
-		self.update_time_counter = cmp::max(self.update_time_counter, header.time);
-		if self.funding_tx_confirmations > 0 {
-			if self.funding_tx_confirmations == self.minimum_depth as u64 {
-				let need_commitment_update = if non_shutdown_state == ChannelState::FundingSent as u32 {
-					self.channel_state |= ChannelState::OurFundingLocked as u32;
-					true
-				} else if non_shutdown_state == (ChannelState::FundingSent as u32 | ChannelState::TheirFundingLocked as u32) {
-					self.channel_state = ChannelState::ChannelFunded as u32 | (self.channel_state & MULTI_STATE_FLAGS);
-					self.update_time_counter += 1;
-					true
-				} else if non_shutdown_state == (ChannelState::FundingSent as u32 | ChannelState::OurFundingLocked as u32) {
-					// We got a reorg but not enough to trigger a force close, just update
-					// funding_tx_confirmed_in and return.
-					false
-				} else if self.channel_state < ChannelState::ChannelFunded as u32 {
-					panic!("Started confirming a channel in a state pre-FundingSent?: {}", self.channel_state);
-				} else {
-					// We got a reorg but not enough to trigger a force close, just update
-					// funding_tx_confirmed_in and return.
-					false
-				};
-				self.funding_tx_confirmed_in = Some(header.block_hash());
-
-				//TODO: Note that this must be a duplicate of the previous commitment point they sent us,
-				//as otherwise we will have a commitment transaction that they can't revoke (well, kinda,
-				//they can by sending two revoke_and_acks back-to-back, but not really). This appears to be
-				//a protocol oversight, but I assume I'm just missing something.
-				if need_commitment_update {
-					if self.channel_state & (ChannelState::MonitorUpdateFailed as u32) == 0 {
-						let next_per_commitment_point = self.holder_signer.get_per_commitment_point(self.cur_holder_commitment_transaction_number, &self.secp_ctx);
-						return Ok((Some(msgs::FundingLocked {
-							channel_id: self.channel_id,
-							next_per_commitment_point,
-						}), timed_out_htlcs));
-					} else {
-						self.monitor_pending_funding_locked = true;
-						return Ok((None, timed_out_htlcs));
-					}
-				}
+			// If we've sent funding_locked (or have both sent and received funding_locked), and
+			// the funding transaction's confirmation count has dipped below minimum_depth / 2,
+			// close the channel and hope we can get the latest state on chain (because presumably
+			// the funding transaction is at least still in the mempool of most nodes).
+			if funding_tx_confirmations < self.minimum_depth as i64 / 2 {
```

Comment:
Is this expected to break test coverage ? 

```
diff --git a/lightning/src/ln/channel.rs b/lightning/src/ln/channel.rs
index dd026398..48602544 100644
--- a/lightning/src/ln/channel.rs
+++ b/lightning/src/ln/channel.rs
@@ -3664,7 +3664,7 @@ impl<Signer: Sign> Channel<Signer> {
                        // the funding transaction's confirmation count has dipped below minimum_depth / 2,
                        // close the channel and hope we can get the latest state on chain (because presumably
                        // the funding transaction is at least still in the mempool of most nodes).
-                       if funding_tx_confirmations < self.minimum_depth as i64 / 2 {
+                       if funding_tx_confirmations < self.minimum_depth as i64 / 4 {
                                return Err(msgs::ErrorMessage {
                                        channel_id: self.channel_id(),
                                        data: format!("Funding transaction was un-confirmed. Locked at {} confs, now have {} confs.", self.minimum_depth, funding_tx_confirmations),
```

Reply:
Hmm, likely not, I'm not sure how much coverage we have for the "confs went down, close channel" stuff, and we should drop it anyway.